qiskit_experiments/experiment_data.py

```diff
@@ -16,16 +16,23 @@
 import uuid
 
 from qiskit.result import Result
-from qiskit.providers import Job, BaseJob
 from qiskit.exceptions import QiskitError
+from qiskit.providers import Job, BaseJob
+from qiskit.providers.exceptions import JobError
 
 
 class AnalysisResult(dict):
     """Placeholder class"""
 
 
 class ExperimentData:
-    """ExperimentData container class"""
+    """ExperimentData container class
+
+    .. warning::
+        This is a temporary data class until the data class is implemented in qiskit terra.
+        Once new data class is implemented, this class will be immediately deprecated.
+        See Qiskit/qiskit-terra#5449.
```

Comment:
5499. I think the plan is to keep this class, and make it inherit from the class in Terra.